src/query/ast/src/parser/statement.rs

```diff
@@ -2090,14 +2075,14 @@ pub fn modify_column_action(i: Input) -> IResult<ModifyColumnAction> {
 
     let modify_column_type = map(
         rule! {
-            #column_type ~ ("," ~ COLUMN ~ #column_type)*
+            #column_def ~ ("," ~ COLUMN ~ #column_def)*
```

Comment:
`column_def` may have computed expr, which is not allowed to modify. Need extra check and report a error.